underscore.js

```diff
@@ -1009,11 +1009,21 @@
 
   // Convert an object into a list of `[key, value]` pairs.
   _.pairs = function(obj) {
-    var keys = _.keys(obj);
-    var length = keys.length;
-    var pairs = Array(length);
-    for (var i = 0; i < length; i++) {
-      pairs[i] = [keys[i], obj[keys[i]]];
+    var pairs, i, length;
+    if (_.isMap(obj)) {
+      var entries = obj.entries();
```

Comment:
Ah right. Ok we may have to hold off a bit more.

Let's be honest though. How useful is this change actually? Are there that
many folks who wouldn't just use Array.from(map.entries())?

On Mon, May 2, 2016, 19:01 Brent Pendergraft <notifications@github.com>
wrote:

> In underscore.js
> <https://github.com/jashkenas/underscore/pull/2523#discussion_r61819425>:
>
> > @@ -1009,11 +1009,21 @@
> >
> >    // Convert an object into a list of `[key, value]` pairs.
> >    _.pairs = function(obj) {
> > -    var keys = _.keys(obj);
> > -    var length = keys.length;
> > -    var pairs = Array(length);
> > -    for (var i = 0; i < length; i++) {
> > -      pairs[i] = [keys[i], obj[keys[i]]];
> > +    var pairs, i, length;
> > +    if (_.isMap(obj)) {
> > +      var entries = obj.entries();
>
> Looks like 4.0.0, which I think was the next official node.js release.
> Everything between the two was io.js.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jashkenas/underscore/pull/2523/files/742033f3de979e46a99d3e9553088e5897d49952#r61819425>
>
-- 

Adam K (mobile)